packages/babel-plugin-transform-block-scoping/src/index.ts

```diff
@@ -84,10 +84,22 @@ export default declare((api, opts: Options) => {
               const { usages, capturedInClosure, hasConstantViolations } =
                 getUsageInBody(binding, path);
 
+              const blockScope = headPath.scope;
+              let varScope = blockScope.getFunctionParent();
+              let isProgramScope = false;
+              if (!varScope) {
+                varScope = blockScope.getProgramParent();
+                isProgramScope = true;
+              }
+
               if (capturedInClosure) {
                 markNeedsBodyWrap();
                 captured.push(name);
-              } else if (headScope.parent.hasBinding(name)) {
+              } else if (
+                headScope.parent.hasBinding(name) ||
+                blockScope.parent.hasGlobal(name) ||
+                (isProgramScope && varScope.hasGlobal(name))
```

Comment:
Since `hasGlobal` checks across all the parent scope, when can it happen that `blockScope.parent.hasGlobal(name)` is false but `isProgramScope && varScope.hasGlobal(name)` is true?

Reply:
I'm not sure, I actually copy-pasted it from here. 🤦‍♂️
Because I found the easiest way is to rename this identifier before `wrapLoopBody`.

https://github.com/babel/babel/blob/112169cc94f70fff6b0fc7641e69be8076dbc90a/packages/babel-plugin-transform-block-scoping/src/index.ts#L213-L231

Reply:
Whops :sweat_smile: 
It looks like deleting the `isProgramScope && varScope.hasGlobal(name)` check also from that code still makes all the tests pass.